metron-analytics/metron-profiler/src/main/java/org/apache/metron/profiler/bolt/ProfileBuilderBolt.java

```diff
@@ -0,0 +1,265 @@
+  }
+
+  @Override
+  public void execute(Tuple input) {
+    try {
+      doExecute(input);
+      collector.ack(input);
+
+    } catch (IOException e) {
+      LOG.error("exception: {}", e);
+      collector.reportError(e);
+    }
+  }
+
+  /**
+   * Update the execution environment based on data contained in the
+   * message.  If the tuple is a tick tuple, then flush the profile
+   * and reset the execution environment.
+   * @param input The tuple to execute.
+   */
+  private void doExecute(Tuple input) throws IOException {
+
+    if(!isTickTuple(input)) {
+
+      // if this is the first tuple in a window period, initialization is needed
+      if (!isInitialized()) {
+        init(input);
+      }
+
+      // update the profile with data from a new message
+      update(input);
+
+    } else {
+
+      // flush the profile - can only flush if it has been initialized
+      if(isInitialized()) {
+        flush(input);
+      }
+    }
+  }
+
+  /**
+   * Initialize the bolt.  Occurs when the first tuple is received at the start
+   * of each window period.
+   * @param input The input tuple
+   */
+  private void init(Tuple input) throws IOException {
+
+    // save the profile definition - needed later during a flush
+    profileConfig = (ProfileConfig) input.getValueByField("profile");
+
+    // create the measurement which will be saved at the end of the window period
+    measurement = new ProfileMeasurement();
+    measurement.setStart(getTimestamp());
+    measurement.setEntity(input.getStringByField("entity"));
+    measurement.setProfileName(profileConfig.getProfile());
+
+    // execute the 'init' expression
+    JSONObject message = (JSONObject) input.getValueByField("message");
+    Map<String, String> expressions = profileConfig.getInit();
+    expressions.forEach((var, expr) -> executor.assign(var, expr, message));
+  }
+
+  /**
+   * Update the Profile based on data contained in a new message.
+   * @param input The tuple containing a new message.
+   */
+  private void update(Tuple input) throws IOException {
+    JSONObject message = (JSONObject) input.getValueByField("message");
+
+    // execute each of the 'update' expressions
+    Map<String, String> expressions = profileConfig.getUpdate();
+    expressions.forEach((var, expr) -> executor.assign(var, expr, message));
+  }
+
+  /**
+   * Flush the Profile.
+   *
+   * Executed on a fixed time period when a tick tuple is received.  Completes
+   * and emits the ProfileMeasurement.  Clears all state in preparation for
+   * the next window period.
+   */
+  private void flush(Tuple tickTuple) throws IOException {
+    LOG.info(String.format("Flushing profile: profile=%s, entity=%s, start=%d",
+            measurement.getProfileName(), measurement.getEntity(), measurement.getStart()));
+
+    // execute the 'result' expression
+    String resultExpr = profileConfig.getResult();
+    Long result = executor.execute(resultExpr, new JSONObject(), Long.class);
```

Comment:
I intended for this to be a Double, not a Long.  We expect the profiler to produce a time series, so a large floating point value is what I would expect to be most useful to users.

On the other hand, do I even need knowledge of what type the resulting value is?  It seems more user-friendly If I just treat it as an Object, and then allow the HBaseBolt to serialize it to HBase.  Then each profile can produce a time series of whatever values they prefer.

The only ugly part is that I'd have to an "if-else instanceof" chain to cast it to the appropriate type in ProfileHBaseMapper.columns(Tuple).  I'd have to do this because Java will not be able to resolve Bytes.toBytes(Object).